dan_layer/core/src/workers/consensus_worker.rs

```diff
@@ -128,7 +134,7 @@ impl<TSpecification: ServiceSpecification<Addr = PublicKey>> ConsensusWorker<TSp
             "Consensus worker started for asset '{}'. Tip: {}", self.asset_definition.public_key, self.current_view_id
         );
         let starting_view = self.current_view_id;
-        loop {
+        while !stop.as_ref().load(Ordering::Relaxed) {
```

Comment:
nit: `.` operator dereferences the `Arc`

```suggestion
        while !stop.load(Ordering::Relaxed) {
```

Reply:
Cool, didn't know that. There is still much to learn in rust for me :)